src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,99 @@
+                             ResponseDtoMapper<ProductResponseDto, Product> responseDtoMapper) {
+        this.productService = productService;
+        this.requestDtoMapper = requestDtoMapper;
+        this.responseDtoMapper = responseDtoMapper;
+    }
+
+    @PostMapping
+    @ApiOperation("Create new product")
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product product = requestDtoMapper.mapToModel(requestDto);
+        return responseDtoMapper.mapToDto(productService.save(product));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation("Get a product by ID")
+    public ProductResponseDto get(@PathVariable Long id) {
+        return responseDtoMapper.mapToDto(productService.get(id));
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation("Update the product by ID and body")
+    public ProductResponseDto get(@PathVariable Long id,
+                                  @RequestBody ProductRequestDto productRequestDto) {
+        Product product = requestDtoMapper.mapToModel(productRequestDto);
+        product.setId(id);
+        return responseDtoMapper.mapToDto(productService.save(product));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation("Delete a product by ID")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @GetMapping
+    @ApiOperation("Get a sorted list of products")
+    public List<ProductResponseDto> findAll(@RequestParam (defaultValue = "0") Integer page,
+                                            @RequestParam (defaultValue = "10") Integer count,
+                                            @RequestParam (defaultValue = "id") String sortBy) {
+        Sort sort = SortUtil.sort(sortBy);
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAll(pageRequest)
+                .stream()
+                .map(responseDtoMapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/price")
+    @ApiOperation("Get a sorted list of products ranged by price")
+    public List<ProductResponseDto> findAll(@RequestParam BigDecimal fromPrice,
+                                            @RequestParam BigDecimal toPrice,
+                                            @RequestParam(defaultValue = "0") Integer page,
+                                            @RequestParam(defaultValue = "10") Integer count,
+                                            @RequestParam(defaultValue = "id") String sortBy) {
+        Sort sort = SortUtil.sort(sortBy);
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAll(pageRequest)
+                .stream()
+                .filter(product -> product.getPrice().doubleValue() >= fromPrice.doubleValue())
+                .filter(product -> product.getPrice().doubleValue() <= toPrice.doubleValue())
```

Comment:
create a separate method on the service layer that will be returning already filtered like you did in the previous task